### PR TITLE
Added ability to correct upload video with a rotation record in the metadata

### PR DESCRIPTION
**Pull request by Marishka17**
### Motivation and context

Ability to correct upload video with a rotation record in the metadata
### How has this been tested?

Manual, tests
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by Marishka17** [comment]
If a video was loaded with a rotation record in the metadata and the 'Use cache' checkbox was enabled, incorrect sizes will be saved. 
This will be fixed after the merge with the [PR](https://github.com/openvinotoolkit/cvat/pull/2218).

**Comment by nmanovic** [comment]
@Marishka17 , what is the status of the PR?

**Comment by Marishka17** [comment]
I left out one moment, that we don't have installed `ffmpeg` inside container.
This was the only way I have found so far how to generate a video with rotation record in the metadata of video stream.
I wrote to PyAV gitter. Maybe someone can suggest another way.

**Review by azhavoro — DISMISSED** [review]
👍

File: cvat/apps/engine/prepare.py [hunk 1]
```diff
@@ -177,6 +195,14 @@ def frame_sizes(self):
         container.seek(offset=next(iter(self.key_frames.values())), stream=video_stream)
         for packet in container.demux(video_stream):
             for frame in packet.decode():
+                if video_stream.metadata.get('rotate'):
+                    frame = av.VideoFrame().from_ndarray(
+                        rotate_image(
+                            frame.to_ndarray(format='bgr24'),
+                            360 - int(container.streams.video[0].metadata.get('rotate'))
```
Comment (nmanovic) [inline cvat/apps/engine/prepare.py:202]: I have a couple of questions here:
- what the method returns: video_stream.metadata.get('rotate')?
- what is the difference between previous expression and container.streams.video[0].metadata.get('rotate')
- can container.streams.video[0].metadata.get('rotate') be always converted to int?

File: cvat/apps/engine/prepare.py [hunk 2]
```diff
@@ -177,6 +195,14 @@ def frame_sizes(self):
         container.seek(offset=next(iter(self.key_frames.values())), stream=video_stream)
         for packet in container.demux(video_stream):
             for frame in packet.decode():
+                if video_stream.metadata.get('rotate'):
```
Comment (nmanovic) [inline cvat/apps/engine/prepare.py:198]: code duplication?

File: cvat/apps/engine/prepare.py (hunk 1 above)
Comment (Marishka17) [reply cvat/apps/engine/prepare.py:202]: > what the method returns: video_stream.metadata.get('rotate')?

string of rotation angle

> what is the difference between previous expression and container.streams.video[0].metadata.get('rotate')

this is analogue

> can container.streams.video[0].metadata.get('rotate') be always converted to int?

I checked for some values like 60, 90, 180, 270.. and in such cases - yes

File: cvat/apps/engine/prepare.py (hunk 2 above)
Comment (Marishka17) [reply cvat/apps/engine/prepare.py:198]: No

File: cvat/apps/engine/utils.py [hunk 3]
```diff
@@ -74,3 +75,16 @@ def av_scan_paths(*paths):
         res = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         if res.returncode:
             raise ValidationError(res.stdout)
+
+def rotate_image(image, angle):
```
[the remark this answers is not in the document]
Comment (Marishka17) [reply cvat/apps/engine/utils.py:79]: Artificially (write the desired value in the metadata) you can create such a video.

**Pull request merged**